lib/formtools/overview.js

```diff
@@ -0,0 +1,145 @@
+
+var async = require('async'),
+    cellRenderers = require('./renderers-cell');
+
+function body(req, res, record, model, overview, cb) {
+
+    // if overview is function execute it and return the content
+    if (typeof overview === 'function') {
+
+        return overview(req, res, record, model, function (err, content) {
+
+            if (err) {
+                return cb(err);
+            }
+
+            return cb(null, {body: content});
+
+        });
+
+    }
+
+    if (!(Array.isArray(overview) && overview.length)) {
+        return cb(null, []);
+    }
+
+    var overviewBody = [];
+
+    // loop through overview array, if body is a fuction, execute it to return the html content,
+    // if  body has fields, render field values,
+    // finally return the overview data
+    async.eachSeries(overview, function (field, callback) {
+
+        if (typeof field !== 'object') {
+            return callback();
+        }
+
+        if (typeof field.body === 'function') {
+
+            return field.body(req, res, record, model, function (err, content) {
+
+                if (err) {
+                    return callback(err);
+                }
+
+                overviewBody.push({
+                    label: field.label,
+                    body: content
+                });
+
+                return callback();
+
+            });
+
+        }
+
+        // make recursive call, if field or field.body is an Array
+        if (Array.isArray(field) || Array.isArray(field.body)) {
```

Comment:
Without being too verbose, can you add a little bit about the DSL you're working with/converting to here? I think this code is a little abstract and could do with a bit more information about what you're working over.